include/revng/Pipeline/ContainerFactory.h

```diff
@@ -22,11 +22,19 @@ class ContainerFactoryBase {
 
   virtual llvm::StringRef mimeType() const = 0;
 
+  virtual std::vector<revng::FilePath>
+  getWrittenFiles(const revng::FilePath &Path) const = 0;
```

Comment:
Maybe this is too powerful.
Maybe we should return `{ "", ".idx" }` so we're never tempted by doing anything else other than appending things to the path.

Reply:
They only meaningful thing one can do with a `revng::FilePath` is to do `addExtension` (other than `getReadableFile`, but that's beside the point). I'm ok to return `std::vector<std::string>`, but that would be a bit less uniform w.r.t. the other methods (also, it might be better to be renamed to `getWrittenExtensions` or something similar)

Reply:
I'll leave this to you. Both solutions sound reasonable to me.

Reply:
I've decided to keep the `revng::FilePath` version for consistency's sake with other functions.